packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -65,31 +65,55 @@ class ChannelCommand extends FlutterCommand {
       <String>['git', 'branch', '-r'],
       workingDirectory: Cache.flutterRoot,
       mapFunction: (String line) {
-        if (verbose) {
-          rawOutput.add(line);
-        }
-        final List<String> split = line.split('/');
-        if (split.length < 2) {
-          return null;
-        }
-        final String branchName = split[1];
-        if (seenChannels.contains(branchName)) {
-          return null;
-        }
-        seenChannels.add(branchName);
-        if (branchName == currentBranch) {
-          return '* $branchName';
-        }
-        if (!branchName.startsWith('HEAD ') &&
-            (showAll || FlutterVersion.officialChannels.contains(branchName))) {
-          return '  $branchName';
-        }
+        rawOutput.add(line);
         return null;
       },
     );
     if (result != 0) {
       final String details = verbose ? '\n${rawOutput.join('\n')}' : '';
       throwToolExit('List channels failed: $result$details', exitCode: result);
+    } else {
```

Comment:
The throwToolExit above will essentially force a return, so you can remove the `else` block to unnest everything here